src/tests/decider-distro-tests-rs/tests/test.rs

```diff
@@ -48,15 +75,402 @@ pub fn make_distro() -> PackageDistro {
+}
+
+// God left me here
+fn modify_decider_spell_script(tmp_dir: PathBuf, decider_spell_id: String, updated_script: String) {
+    let script_path: PathBuf = tmp_dir.join(
+        [
+            "services",
+            "workdir",
+            &decider_spell_id,
+            "tmp",
+            "script.air",
+        ]
         .iter()
-        .map(|s| s.name.clone())
-        .chain(distro.spells.iter().map(|s| s.name.clone()))
+        .collect::<PathBuf>(),
+    );
+
+    fs::write(&script_path, updated_script).unwrap();
+}
+
+async fn update_decider_script_for_tests(client: &mut ConnectedClient, test_dir: PathBuf) {
+    let result = execute(
+        client,
+        r#"
+            (seq
+                (call relay ("srv" "resolve_alias_opt") ["decider"] id)
+                (call relay ("decider" "get_script_source_from_file") [] script)
+            )
+    "#,
+        "id script",
+        hashmap! {},
+    )
+    .await
+    .unwrap();
+    assert_eq!(
+        result[0].as_array().unwrap().len(),
+        1,
+        "can't resolve `decider` alias"
+    );
+    let decider_id = result[0].as_array().unwrap()[0]
+        .as_str()
+        .unwrap()
+        .to_string();
+    let script = serde_json::from_value::<ScriptValue>(result[1].clone()).unwrap();
+    assert!(script.success, "can't get decider script");
+
+    let updated_script = format!(
+        r#"
+        (seq
+            {script}
+            (call "{client}" ("return" "") ["done"])
+        )
+    "#,
+        client = client.peer_id,
+        script = script.source_code,
+    );
+
+    modify_decider_spell_script(test_dir, decider_id, updated_script);
+}
+
+fn oneshot_config() -> TriggerConfig {
+    let mut config = TriggerConfig::default();
+    config.clock.start_sec = 1;
+    config
+}
+
+async fn wait_worker_spell(
+    client: &mut ConnectedClient,
+    worker_id: String,
+    timeout_per_try: Duration,
+) {
+    for step in 0..5 {
+        // if only we can import these keys from Aqua files
+        let result = execute(
+            client,
+            r#"
+            (seq
+                (call relay ("op" "noop") [])
+                (call worker ("worker-spell" "list_get_strings") ["__installation_spell_status__"] status)
+            )
+        "#,
+            r#"status"#,
+            hashmap! {
+                "worker" => json!(worker_id),
+            },
+        )
+        .await
+        .unwrap();
+        assert!(!result.is_empty());
+        let strings = serde_json::from_value::<StringListValue>(result[0].clone()).unwrap();
+        assert!(strings.success);
+        if !strings.strings.is_empty() {
+            #[derive(Deserialize, Debug)]
+            struct State {
+                state: String,
+            }
+            let last_status = strings.strings.last().unwrap();
+            let state = serde_json::from_str::<State>(last_status).unwrap();
+            if state.state != "INSTALLATION_IN_PROGRESS" {
```

Comment:
Wouldn't that fail if called too early, when status is not yet `INSTALLATION_IN_PROGRESS`?

Maybe have a match over `state.state` and accept all non-error statuses?